.github/workflows/prepare_release.yml

```diff
@@ -0,0 +1,53 @@
+name: Prepare release
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  prepare_release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set RELEASE_VERSION
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Prepare release
+
+      - name: Setup Python 3.8
+        uses: actions/setup-python@v2
+        with: python-version:3.8
+
+      - name: Python Poetry Action
+        uses: abatilo/actions-poetry@v2.1.0
+
+      - name: Install dependencies
+        run: poetry install
+
+      - name: Get current version
+        run: echo "CURRENT_VERSION=$(poetry version | awk '{print $2}')" >> $GITHUB_ENV
+
+      - name: Generate changelog
+        run: |
+          poetry run tools/gen_changelog.py ${{ secrets.GITHUB_TOKEN }} v$CURRENT_VERSION v$RELEASE_VERSION
+
+      - name: Bump version
+        run: poetry version $RELEASE_VERSION
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
```

Comment:
I think it's a best practice to use a commit SHA instead of a tag (e.g. `v2`). Tags can be deleted (so they're not guaranteed to be there, although this rarely happens), and they can be re-added to another commit, which might introduce security and reproducibility issues. Not sure how crucial it for Maha, but thought I'd point it out.

Reply:
Thank you for your comment. I am not sure if this is possible to be triggered automatically. Tags are used here only to trigger the workflow. They can also be helpful later to take a snapshot of the code once a new release is published. I believe most of the open-source projects use tags for versioning, which can also be sufficient for Maha as I don't think we will ever need to re-assign a version tag. Would you mind sharing few resources on how commit SHA are used for versioning or projects that use it instead of tags?

Reply:
Oh, sorry, I might've not been clear.
I was talking more about the github actions references in `uses: ...@...`. The current approach of using `@v3` is valid and shouldn't be a problem generally. I was pointing out that referencing a version of a github action by its commit SHA is safer. I wasn't referring to Maha's versioning.

Reference: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsuses

Reply:
Sorry for the misunderstanding. I do agree that commit SHA is safer, but I'd rather use the major version method instead to receive critical fixes and security patches automatically.

Note that I haven't been able to test the workflow and I am not sure if it does what it's supposed to do. Would love to hear more comments on it, if possible.